routes/routes.js

```diff
@@ -95,25 +96,31 @@ router.get('/products', (req, res) => {
   }
 });
 
-router.get('/products/:productId', (req, res) => {
+router.get('/products/:productId', (req, res, next) => {
   try {
     const productId = parseInt(req.params.productId);
     const product = products.find((product) => product.id === productId);
-    if (!product) return res.status(404).json({ error: 'Product not found' });
+    if (!product) {
+      const err = new Error('Product not found');
```

Comment:
`Product not found`
This error message will be overridden in the error handler middleware. Fix it.